### ISSUE TITLE
numpy binary incompatibility for pandas 0.23.4 and numpy 1.15 with pip and .whl

### ISSUE THREAD

**Issue by garaud** [pandas-dev/pandas#22444]
#### Code Sample, a copy-pastable example if possible

A simple pandas import

```python
import pandas as pd
```

#### Problem description

Several `RuntimeWarnings` in Python 2 or Python 3. I think it's due to a different version of numpy between:

* the version of numpy used to compile pandas and built the `.whl` file available on Pypi https://pypi.org/project/pandas/#files
* the latest version of numpy on Pypi, i.e. 1.15

I think the only version constraint is `numpy >= 1.9`. When you install with pip, it gets the latest version of numpy.

I wonder how I can retrieve the version of numpy used to compile pandas and build the `.whl` file. If I reinstall `numpy==1.14.5`, I don't have the `RuntimeWarning`.

#### Outputs

```
venv/local/lib/python2.7/site-packages/pandas/_libs/__init__.py:4: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from .tslib import iNaT, NaT, Timestamp, Timedelta, OutOfBoundsDatetime
venv/local/lib/python2.7/site-packages/pandas/__init__.py:26: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from pandas._libs import (hashtable as _hashtable,
venv/local/lib/python2.7/site-packages/pandas/core/dtypes/common.py:6: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from pandas._libs import algos, lib
venv/local/lib/python2.7/site-packages/pandas/core/util/hashing.py:7: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  from pandas._libs import hashing, tslib
venv/local/lib/python2.7/site-packages/pandas/core/indexes/base.py:7: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
...
and other `_libs` imports
```

#### Output of ``pd.show_versions()``

<details>

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.15.final.0
python-bits: 64
OS: Linux
OS-release: 4.16.0-2-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.23.4
pip: 18.0
setuptools: 40.2.0
numpy: 1.15.0
IPython: 5.8.0
dateutil: 2.7.3
pytz: 2018.5
```

</details>

**Comment by garaud** [comment]
I quickly read the numpy [1.15 Relase Notes](https://docs.scipy.org/doc/numpy/release.html) and I found https://github.com/numpy/numpy/pull/10898 about user-defined dtypes.

**Comment by TomAugspurger** [comment]
pandas 2.7  wheels are built with NumPy 1.9.3. Here's the log: https://travis-ci.org/MacPython/pandas-wheels/jobs/412100528

Why do you think https://github.com/numpy/numpy/pull/10898 is releated?

**Comment by garaud** [comment]
Thank you for the link. I also found this https://github.com/MacPython/pandas-wheels/blob/master/.travis.yml I've the same info as the travis CI website.

About https://github.com/numpy/numpy/pull/10898, it's just a hunch. I thought that PR modified a C struct related to dtype which would have changed the size of `numpy.dtype`. But, it's a function, so... wrong hunch I guess. That was the only C API changes related to dtype from the release notes.

I've got this warning on Debian Sid. I also reproduce it on CentOS 7, in a python2 or python3 virtualenv.

**Comment by garaud** [comment]
I found other hints

* https://docs.scipy.org/doc/numpy/release.html#allow-dtype-field-names-to-be-unicode-in-python-2
* https://github.com/numpy/numpy/commit/16f39c80f69ae4695f2f940ccdff8b26db3a2a01

**Comment by jorisvandenbossche** [comment]
This should be fixed in numpy 1.15.1, just released. Quoting from the release notes:

> The annoying but harmless RuntimeWarning that "numpy.dtype size changed" has
been suppressed. The long standing suppression was lost in the transition to
pytest.

See https://github.com/numpy/numpy/releases/tag/v1.15.1

**Comment by garaud** [comment]
Thanks @jorisvandenbossche for the news. With this version of numpy, RuntimeWarnings were vanished. If you know the numpy commit which fixed this stuff, I'm curious. Thanks.
